### PR TITLE
doc: zigbee: expand default signal handler docs

### PR DESCRIPTION
**Pull request by greg-fer**
Added new images to Zigbee default signal handler docs.
Applied minor style edits.
KRKNWK-10712.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

File: include/zigbee/zigbee_app_utils.rst [hunk 1]
```diff
@@ -277,29 +282,49 @@ It is used in :c:func:`zigbee_default_signal_handler` by default.
...
 
-The period is limited to 15 minutes if the result is higher than that.
+When :c:func:`start_network_rejoin` is called, the rejoin procedure is started, and depending on the device role:
 
-* When :c:func:`start_network_rejoin` is called, the rejoin procedure is started, and depending on the device role:
+When ``stop_network_rejoin(was_scheduled)`` is called, the network rejoin is canceled and the alarms scheduled by :c:func:`start_network_rejoin` are canceled.
 
-  * For the end device, the application alarm is scheduled with ``stop_network_rejoin(ZB_TRUE)``, to be called after the amount of time specified in ``ZB_DEV_REJOIN_TIMEOUT_MS``.
-    Once called, the alarm stops the rejoin.
+  * Additionally for the end device, if :c:func:`stop_network_rejoin` is called with ``was_scheduled`` set to ``ZB_TRUE``, :c:func:`user_input_indicate` can restart the rejoin procedure.
```
Comment (tomchy) [inline include/zigbee/zigbee_app_utils.rst:297]: This sentence is in the `Router rejoining network` section. If it is applicable for the router devices - I think the condition (`for the end device`) should be removed. If it is not - why it was not moved completely to the `End device rejoining network` section?

File: include/zigbee/zigbee_app_utils.rst [hunk 2]
```diff
@@ -277,29 +282,49 @@ It is used in :c:func:`zigbee_default_signal_handler` by default.
...
+Once called, the alarm stops the rejoin.
+
+When ``stop_network_rejoin(was_scheduled)`` is called, the network rejoin is canceled and the alarms scheduled by :c:func:`start_network_rejoin` are canceled.
+If :c:func:`stop_network_rejoin` is called with ``was_scheduled`` set to ``ZB_TRUE``, :c:func:`user_input_indicate` can restart the rejoin procedure.
+:c:func:`user_input_indicate` restarts the rejoin procedure if the device did not join the network and is not trying to join a network.
+It is safe to call this function from an interrupt and to call it multiple times.
 
-* For end devices only, :c:func:`user_input_indicate` restarts the rejoin procedure if the device did not join the network and is not trying to join a network.
-  It is safe to call this function from an interrupt and to call it multiple times.
+.. figure:: /images/zigbee_signal_handler_10_rejoin_end_device.svg
```
Comment (tomchy) [inline include/zigbee/zigbee_app_utils.rst:324]: I think this figure needs to be improved: the `User action` is not part of a regular flow, meaning you do not check for user action after the whole rejoin procedure stops. This should be rather presented as an asynchronous event.

File: include/zigbee/zigbee_app_utils.rst [hunk 3]
```diff
@@ -135,7 +135,12 @@ Complete zigbee_default_signal_handler implementation
 
 In its complete implementation, the `zboss_signal_handler()`_ allows the application to control a broader set of basic functionalities, including joining, commissioning, and network formation.
 
-The following sections describe the logic implemented inside :c:func:`zigbee_default_signal_handler`.
+.. figure:: /images/zigbee_signal_handler_overview.svg
```
Comment (tomchy) [inline include/zigbee/zigbee_app_utils.rst:138]: I do not understand this figure: so there are four options after the BDB initialization (new device scenario, commissioned device scenario, network joining, network formation) ?
Is not it that way, that the scenarios are enclosed inside the same block as the BDB initialization?

Comment (tomchy) [reply include/zigbee/zigbee_app_utils.rst:138]: I'm not sure if I am a fan of arrows partially hidden behind boxes

File: include/zigbee/zigbee_app_utils.rst [hunk 4]
```diff
@@ -135,7 +135,12 @@ Complete zigbee_default_signal_handler implementation
...
+   :alt: Zigbee default signal handler logic (simplified)
+
+   Zigbee default signal handler logic (simplified)
+
+The following sections describe the logic implemented inside :c:func:`zigbee_default_signal_handler` and correspond to the shapes on the figure.
```
Comment (Avalei) [inline include/zigbee/zigbee_app_utils.rst:143]:
```suggestion
The following sections describe the logic implemented inside :c:func:`zigbee_default_signal_handler` and correspond to the shapes in the figure.
```

**Review by Avalei — CHANGES_REQUESTED** [review]
doc/nrf/images/zigbee_signal_handler_10_rejoin.svg
- The "false" text has a white fill color.

doc/nrf/images/zigbee_signal_handler_10_rejoin_user_input.svg
- Since there is plenty of room, it would look more readable to have (eg. button press) all on one line). Keep all text in the box center aligned.

doc/nrf/images/zigbee_signal_handler_overview.svg
- Thinking about this some more, I wonder if we should have the Network start and Network leaving states in the lightest gray color with dark text so that it is not misinterpreted as being part of the Zigbee stack sleep routines.

**Review by tomchy — APPROVED** [review]

**Review by sebastiandraus — APPROVED** [review]

**Review by Avalei — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**